### PR TITLE
west.yml: Update Mbed TLS to 2.26.0

### PR DESCRIPTION
**Pull request by d3zd3z**
Move to the latest release of Mbed TLS.

Signed-off-by: David Brown <david.brown@linaro.org>

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -88,7 +90,8 @@ manifest:
       revision: 31acbaa36e9e74ab88ac81e3d21e7f1d00a71136
       path: modules/lib/gui/lvgl
     - name: mbedtls
-      revision: 53044168b43f06ec5654b906a2cdd260bf477edd
+      remote: d3zd3z
```
Comment (nashif) [inline west.yml:93]: you should be pointing to the PR in the mbedtls module , not some private remote:

`revision: pull/23/head`

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by frkv** [comment]
Adding mbed TLS v2.26.0 is not compatible with TF-M 1.3 (latest release) or TF-M master

The latest current release used in the TF-M project is mbed TLS v2.25

**Comment by microbuilder** [comment]
> Adding mbed TLS v2.26.0 is not compatible with TF-M 1.3 (latest release) or TF-M master
> 
> The latest current release used in the TF-M project is mbed TLS v2.25.0

TF-M builds for the secure environment should continue to pull in it's own copy of mbedtls, which will reference 2.25 as of TF-M 1.3.0: https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/master/trusted-firmware-m/config/config_default.cmake#L164

At present, this module is for use with the NS environment, enabling use of the PSA APIs as a backend in mbedtls. Initial testing seems to indicate that 2.26.0 on the NS side is still compatible with TF-M 1.3.0, but if you have seen any specific issues please bring them up so that we can investigate.

There is a goal to bring the entire mbedtls history into Zephyr's mbedtls module repo, which will allow us to reference past tags/releases as well, but that's still a WIP, and 2.26.0 was selected since it is currently the latest release.

**Comment by ioannisg** [comment]
I guess it is possible to force TF-M builds in Zephyr to use the mbedtls 2.26 as well. It is a configuration, really, so out-of-tree projects can re-modify this to use the "DOWNLOAD" option.

**Comment by ceolin** [comment]
> Adding mbed TLS v2.26.0 is not compatible with TF-M 1.3 (latest release) or TF-M master
> 
> The current release used in the TF-M project is mbed TLS v2.25.0

Yep, TF-M is not the only reason. The other thing that motivated moving to the latest release is that end of life for the current LTS is this year  and this does not work for next Zephyr LTS. Moving to something newer right now allow us find possible failures before next  Zephyr LTS release.

**Review by microbuilder — APPROVED** [review]
Unable to approve the PR on the mbedtls repo, but tested and seems OK here (so far).

**Comment by ioannisg** [comment]
> Unable to approve the PR on the mbedtls repo, but tested and seems OK here (so far).

The crypto.mbedtls test seems to be failing, though

**Review by ioannisg — APPROVED** [review]

**Comment by microbuilder** [comment]
@frkv Just as a FYI, I did a test pointing TF-M on the S side to mbedtls 2.26.0 and it still builds and runs as expected.

**Review by ceolin — APPROVED** [review]

**Pull request closed without merging**